### PR TITLE
Race condition on harvester limit

### PR DESCRIPTION
**Pull request by adriansr**

### CODE REVIEW

**Comment by tsg** [comment]
Cross-linking #5458.

File: filebeat/harvester/registry.go [hunk 1]
```diff
@@ -5,6 +5,8 @@ import (
 
 	uuid "github.com/satori/go.uuid"
 
+	"fmt"
```
Comment (andrewkroh) [inline filebeat/harvester/registry.go]: Please move this up to the top with the other stdlib imports.

File: filebeat/tests/system/test_prospector.py [hunk 2]
```diff
@@ -583,7 +583,7 @@ def test_skip_symlinks(self):
         # Make sure there is only one entry, means it didn't follow the symlink
         assert len(data) == 1
 
-    @unittest.skip("Flaky due to race. Opened https://github.com/elastic/beats/issues/5458")
+    #@unittest.skip("Flaky due to race. Opened https://github.com/elastic/beats/issues/5458")
```
Comment (andrewkroh) [inline filebeat/tests/system/test_prospector.py]: I think this can be removed since it should be fixed.

File: filebeat/harvester/registry.go [hunk 3]
```diff
@@ -59,14 +60,14 @@ func (r *Registry) WaitForCompletion() {
...
-func (r *Registry) Start(h Harvester) {
+func (r *Registry) Start(h Harvester) error {
 	// Make sure stop is not called during starting a harvester
 	r.Lock()
 	defer r.Unlock()
 
 	// Make sure no new harvesters are started after stop was called
 	if !r.active() {
-		return
+		return fmt.Errorf("registry already stopped")
```
Comment (urso) [inline filebeat/harvester/registry.go]: No formatting, use `errors.New` + (optional here) predefined variable.

**Pull request merged**